native/cocos/platform/interfaces/modules/ISystemWindowManager.h

```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+#include "platform/interfaces/OSInterface.h"
+
+namespace cc {
+class ISystemWindow;
+class OSEvent;
+
+struct ISystemWindowInfo {
+    ccstd::string title{""};
+    int32_t x{-1};
+    int32_t y{-1};
+    int32_t width{-1};
+    int32_t height{-1};
+    int32_t flags{-1};
+    void *externalHandle{nullptr};
+};
+
+typedef ccstd::unordered_map<uint32_t, std::shared_ptr<ISystemWindow>> SystemWindowMap;
+
+/**
+ * 负责创建、查找 ISystemWindow 及消息处理
+ */
+class ISystemWindowManager : public OSInterface {
+public:
+    /**
+     * 初始化 NativeWindow 环境
+     * @return 0 成功 -1 失败
+     */
+    virtual int init() = 0;
+
+    /**
+     * 处理 PAL 层的消息
+     */
+    virtual void processEvent(bool *quit) = 0;
```

Comment:
Why use pointer for bool?

Reply:
It is an out parameter.

Reply:
Why need an out parameter?

Reply:
Use it to judge whether the engine triggered an exit event.

Reply:
Why not return a bool value, such as 
```c++
virtual bool processEvent() = 0;
```

Reply:
In general, the return value represents the result of the function, which may be failure, success, or an error code.

Because the return value of 'quit' is the result returned inside the nested function, it cannot represent the execution result of the function. So I think it's better to use function parameter return

Reply:
In general, the return value represents the result of the function, which may be failure, success, or an error code.

Because the return value of 'quit' is the result returned inside the nested function, it cannot represent the execution result of the function. So I think it's better to use function parameter return